### DISCUSSION TITLE
Mutation category levels - seems to be looking exclusively at whether THIS character has one as a base/starting trait?

### DISCUSSION THREAD

**Discussion by actual-nh** [CleverRaven/Cataclysm-DDA#48172] [Q&A]
From character.cpp:
```cpp
void Character::build_mut_dependency_map( const trait_id &mut,
        std::unordered_map<trait_id, int> &dependency_map, int distance )
{
    // Skip base traits and traits we've seen with a lower distance
    const auto lowest_distance = dependency_map.find( mut );
    if( !has_base_trait( mut ) && ( lowest_distance == dependency_map.end() ||
                                    distance < lowest_distance->second ) ) {
        dependency_map[mut] = distance;
        // Recurse over all prerequisite and replacement mutations
        const mutation_branch &mdata = mut.obj();
        for( const trait_id &i : mdata.prereqs ) {
            build_mut_dependency_map( i, dependency_map, distance + 1 );
        }
        for( const trait_id &i : mdata.prereqs2 ) {
            build_mut_dependency_map( i, dependency_map, distance + 1 );
        }
        for( const trait_id &i : mdata.replacements ) {
            build_mut_dependency_map( i, dependency_map, distance + 1 );
        }
    }
}
```
The function `has_base_trait` is not checking for whether something **_is_** a possible starting trait; it's checking for whether **_this character_** has that trait (and not through mutation?), as far as I can tell. Why? (If my thinking is correct, this means that not only do starting traits that are in a category not push you toward that category, they make it **_harder_** to get that category, possibly counterbalanced by being a prerequisite... at least, if they're a prerequisite for mutations in **_that_** category - which is frequently not true - and if the prerequisite isn't satisfied by an alternative.)

In other words, two questions:

- Is my thinking correct re the effects of the code?
- Are these effects intentional, or is this a bug?

Ping: @I-am-Erk, @Venera3?
